src/Polysemy/Internal/Forklift.hs

```diff
@@ -66,7 +67,7 @@ withLowerToIO action = do
   res <- embed $ A.async $ do
     a <- action (runViaForklift inchan)
                 (putMVar signal ())
-    putMVar signal ()
+          `finally` (putMVar signal ())
```

Comment:
Pointing this out so you won't miss it; this fixes the "exception" part of #205 .